x/programs/rust/examples/nft/src/lib.rs

```diff
@@ -0,0 +1,141 @@
+//! A basic ERC-721 compatible contract.
+//! The program serves as a non-fungible token with the ability to mint and burn.
+//! Only supports whole units with no decimal places.
+//!
+//! The NFT must support the common NFT metadata format.
+//! This includes the name, symbol, and URI of the NFT.
+use metadata::Nft;
+use wasmlanche_sdk::{program::Program, public, state_keys, types::Address};
+
+pub mod example;
+pub mod metadata;
+
+const NAME: &str = "My NFT";
+const SYMBOL: &str = "MNFT";
+const TOTAL_SUPPLY: u64 = 20;
+
+/// The program storage keys.
+#[state_keys]
+enum StateKey {
+    /// The total supply of the token. Key prefix 0x0.
+    TotalSupply,
+    /// The name of the token. Key prefix 0x1.
+    Name,
+    /// The symbol of the token. Key prefix 0x2.
+    Symbol,
+    /// Metadata of the token. Key prefix 0x3.
+    Metadata,
+    /// Balance of the NFT token by address. Key prefix 0x5(address).
+    Balance(Address),
+    /// Counter -- used to keep track of total NFTs minted.
+    Counter,
+}
+
+/// Initializes the NFT with all required metadata.
+/// This includes the name, symbol, image URI, owner, and total supply.
+/// Returns true if the initialization was successful.
+#[public]
+pub fn init(program: Program) -> bool {
+    // Set token name
+    program
+        .state()
+        .store(StateKey::Name.to_vec(), &NAME.as_bytes())
+        .expect("failed to store nft name");
+
+    // Set token symbol
+    program
+        .state()
+        .store(StateKey::Symbol.to_vec(), &SYMBOL.as_bytes())
+        .expect("failed to store nft symbol");
+
+    // Generate NFT metadata and persist to storage
+    let nft_metadata = Nft::default()
+        .with_symbol(SYMBOL.to_string())
+        .with_name(NAME.to_string())
+        .with_uri("ipfs://my-nft.jpg".to_string());
+
+    program
+        .state()
+        .store(StateKey::Metadata.to_vec(), &nft_metadata)
+        .expect("failed to store nft metadata");
+
+    // Set total supply
+    program
+        .state()
+        .store(StateKey::TotalSupply.to_vec(), &TOTAL_SUPPLY)
+        .expect("failed to store total supply");
+
+    true
+}
+
+/// Mints NFT tokens and sends them to the recipient.
+#[public]
+pub fn mint(program: Program, recipient: Address) -> bool {
+    let mint_amount: i64 = 1;
```

Comment:
```suggestion
    const MINT_AMOUNT: i64 = 1;
```
?

otherwise, should be
```suggestion
    let mint_amount = 1_i64;
```